versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/FetchValuesUsingOpenAddressing.java

```diff
@@ -90,17 +89,19 @@ List<Hash> entityIdsToFetch(int round, int prefetchEntities, Collection<Key> rem
     // Identify the key-list segments to fetch
     List<Hash> entitiesToFetch = new ArrayList<>();
     for (Key key : remainingKeys) {
-      int keyBucket = segmentForKey(key);
+      int keyBucket = bucketForKey(key);
       int segment = segmentForKey(keyBucket, round);
 
       for (int prefetch = 0; ; prefetch++) {
         if (segment > 0) {
+          checkSegmentIndexValidity(segment, key);
+
+          // Decrement accounts for the embedded key-list segment
           int entitySegment = segment - 1;
-          if (keyListIds.size() > entitySegment) {
-            Hash entityId = keyListIds.get(entitySegment);
-            if (entityIdToSegment.put(entityId, segment) == -1) {
-              entitiesToFetch.add(entityId);
-            }
+
+          Hash entityId = keyListIds.get(entitySegment);
+          if (entityIdToSegment.put(entityId, segment) == -1) {
+            entitiesToFetch.add(entityId);
```

Comment:
I'm removing the `keyListIds.size() > entitySegment` condition because I'm now relying on `segmentForKey` to ensure that property (wrapping if necessary), and confirming it just above in `checkSegmentIndexValidity`.